lib/firmata.js

```diff
@@ -350,6 +358,21 @@ SYSEX_RESPONSE[STEPPER] = function(board) {
   board.emit("stepper-done-" + deviceNum, true);
 };
 
+SYSEX_RESPONSE[SERIAL_MESSAGE] = function(board) {
+  var command = board.currentBuffer[2] & 0xF0;
+  var portId = board.currentBuffer[2] & 0x0F;
+  var data;
+
+  if (command === (SERIAL_REPLY)) {
+    for (var i = 3, len = board.currentBuffer.length; i < len - 1; i += 2) {
+      // better to send one byte at a time or entire buffer?
+      // currently sending just one byte at a time
+      data = (board.currentBuffer[i + 1] << 7) | board.currentBuffer[i];
+      board.emit("serial-data", {data: data, portId: portId});
```

Comment:
The portId would be needed or the user wouldn't know which port the data corresponds to. Or are you suggesting:
```javascript
board.emit("serial-data", data, portId);

// or

board.emit('serial-data-" + portId, data); // seems to be the pattern but my least preferred
```